src/hl/dataset.rs

```diff
@@ -456,13 +457,14 @@ impl DatasetBuilderInner {
         let chunking_required = has_filters || extents.is_resizable();
         let chunking_allowed = extents.size() > 0 || extents.is_resizable();
 
-        let chunk = if let Some(chunk) = &self.chunk {
-            chunk.clone()
-        } else if chunking_required && chunking_allowed {
-            Chunk::MinKB(DEFAULT_CHUNK_SIZE_KB)
-        } else {
-            Chunk::None
-        };
+        let chunk = self.chunk.as_ref().map_or(
```

Comment:
Same as with `.map_or_else()` case above, not sure I agree this is more readable? (I'd probably ignore it along with map-or-else)

Reply:
Blacklisted and reverted this code